src/shim.c

```diff
@@ -331,17 +331,17 @@ send_connect_command(struct cc_shim *shim)
  *
  * \return true on success, false otherwise
  */
-bool read_wire_data(int fd, uint8_t *buf, ssize_t size)
+bool read_wire_data(int fd, uint8_t *buf, size_t size)
 {
 	ssize_t ret;
-	ssize_t offset = 0;
+	size_t offset = 0;
 
-	if ( fd < 0 || ! buf ) {
+	if ( fd < 0 || ! buf) {
 		return false;
 	}
 
-	do {
-		ret = recv(fd, buf, (size_t)(size-offset), 0);
+	while(offset < size) {
+		ret = recv(fd, buf+offset, size-offset, 0);
```

Comment:
There are two things. The refactoring (including the loop refactor) is one commit. Then there's the `buf+offset`, which is not part of the refactoring as far as I can see (but could be wrong). That's a second commit. But I could be wrong of course.